Add String-based JSON API to avoid unnecessary conversions (#3369)

KoKimSS opened the pull request:
This PR adds String-based overloads for JSON commands to eliminate unnecessary conversion overhead between String ↔ JsonValue objects, addressing performance concerns raised in the feature request #3369.

**Important Note**

The existing JSON API assumes users only work with `JsonValue` objects. In certain cases, applications might want to provide a String interface instead, because otherwise they'd have to convert from string to `JsonValue` and then the driver would convert back to String and then to a byte array.

This approach provides String-based overloads that work directly with String input/output parameters, instead of `JsonValue` objects, eliminating unnecessary conversions while maintaining full backward compatibility.

**Performance Benefits**
* **Before**: `String → JsonValue.of() → toString() → getBytes() → ByteBuffer.wrap() → array()`
* **After**: `String → getBytes()` (direct conversion)
* **Result**: Eliminates unnecessary object creation and conversion steps

## Changes Made

### New String Overload Methods Added
* `jsonArrappend(K key, JsonPath jsonPath, String... jsonStrings)`
* `jsonArrindex(K key, JsonPath jsonPath, String jsonString, JsonRangeArgs range)`  
* `jsonArrinsert(K key, JsonPath jsonPath, int index, String... jsonStrings)`
* `jsonMerge(K key, JsonPath jsonPath, String jsonString)`
* `jsonSet(K key, JsonPath jsonPath, String jsonString, JsonSetArgs options)`
* `jsonStrappend(K key, JsonPath jsonPath, String jsonString)`

### Core Implementation
* **RedisJsonCommandBuilder**: Added String overload methods with direct string-to-bytes conversion
* **RedisJsonAsyncCommands**: Updated async interface with String overloads
* **RedisJsonReactiveCommands**: Updated reactive interface with String overloads  
* **RedisJsonCommands**: Updated sync interface with String overloads
* **AbstractRedisAsyncCommands**: Added implementation for all String overloads
* **AbstractRedisReactiveCommands**: Added reactive implementations

### Testing
* **RedisJsonCommandBuilderUnitTests**: Added comprehensive unit tests for all String overload methods
* Command encoding verification for both custom JsonPath and ROOT_PATH scenarios
* Direct string encoding optimization validation

KoKimSS committed:
Add String-based JSON API to avoid unnecessary conversions #3369

- Add String overloads for jsonArrappend, jsonArrindex, jsonArrinsert
- Add String overloads for jsonMerge, jsonSet, jsonStrappend
- Implement direct String-to-bytes conversion in RedisJsonCommandBuilder
- Update Async, Reactive, and Sync interfaces consistently
- Eliminate String → JsonValue → String → bytes conversion overhead

KoKimSS committed:
Add comprehensive tests for String JSON API #3369

- Add unit tests for all String overload methods
- Verify direct string encoding optimization
- Test both custom JsonPath and ROOT_PATH scenarios
- Validate command construction for all String-based JSON operations

injae-kim commented on src/main/java/io/lettuce/core/RedisJsonCommandBuilder.java:
```diff
@@ -256,6 +353,32 @@ Command<K, V, String> jsonSet(K key, JsonPath jsonPath, JsonValue value, JsonSet
 
+    /**
+     * JSON.SET command with raw string value to avoid unnecessary conversions.
+     *
+     * @param key      the key
```
maybe we can just use 1 space here? can you check comment on other files?
```suggestion
     * @param key the key
```

KoKimSS replied:
Thanks for the review!

First, I'll remove the `@param` documentation from the `RedisJsonCommandBuilder` methods since I noticed that other methods in this class don't have JavaDoc comments - they follow the pattern of keeping the documentation only in the interface level.

Second, I checked the other files where I added `@param` documentation and they already follow the correct single space pattern you mentioned.

KoKimSS committed:
Polish JSON API documentation and formatting #3369

- Remove JavaDoc comments from RedisJsonCommandBuilder String methods to match class conventions
- Update @author entries to use full names (SeungSu Kim)
- Fix @since annotation from 6.5 to 6.9 for new String overload methods
- Restore blank lines and fix spacing to maintain consistent formatting
- Align documentation style with existing codebase patterns

a-TODO-rov committed:
CAE-1404 - finish String extension for JSON API

a-TODO-rov committed:
Fix since annotation on json string API

a-TODO-rov committed:
Rm useless comments in implementations and mark methods as Override

a-TODO-rov committed:
Rm more comments.

a-TODO-rov committed:
Format
